packages/client/ui/react-ui/src/providers/auth/web3/DynamicWeb3WalletConnect.tsx

```diff
@@ -0,0 +1,73 @@
+import type React from "react";
+import { EthereumWalletConnectors } from "@dynamic-labs/ethereum";
+import { SolanaWalletConnectors } from "@dynamic-labs/solana";
+import type { APIKeyEnvironmentPrefix } from "@crossmint/common-sdk-base";
+import DynamicContextProviderWrapper from "@/components/dynamic-xyz/DynamicContextProviderWrapper";
+import { useCrossmintAuth } from "@/hooks/useCrossmintAuth";
+import { useAuthForm } from "../AuthFormProvider";
+import { dynamicChainToCrossmintChain } from "@/utils/dynamic/dynamicChainToCrossmintChain";
+
+export function DynamicWeb3WalletConnect({
+    children,
+    apiKeyEnvironment,
+}: { children: React.ReactNode; apiKeyEnvironment: APIKeyEnvironmentPrefix }) {
+    const { crossmintAuth } = useCrossmintAuth();
+    const { appearance } = useAuthForm();
+
+    const cssOverrides = `.powered-by-dynamic { display: none !important; } .wallet-list__scroll-container { padding: 0px !important; } .wallet-list__search-container { padding-left: 0px !important; padding-right: 0px !important; } .dynamic-footer { display: none !important; } h1 { color: ${appearance?.colors?.textPrimary} !important; } * { color: ${appearance?.colors?.textSecondary} !important; }`;
+
+    return (
+        <DynamicContextProviderWrapper
+            apiKeyEnvironment={apiKeyEnvironment}
+            settings={{
+                cssOverrides,
+                walletConnectors: [EthereumWalletConnectors, SolanaWalletConnectors],
+                events: {
+                    onAuthFlowCancel() {
+                        console.log("[CryptoWalletConnectionHandler] onAuthFlowCancel");
+                    },
+                    onAuthFlowClose() {
+                        console.log("[CryptoWalletConnectionHandler] onAuthFlowClose");
+                    },
+                    onAuthFailure(data, reason) {
+                        console.error("[CryptoWalletConnectionHandler] onAuthFailure", data, reason);
```

Comment:
shouldnt we use our new error handling here?

Reply:
Yes, i'll make a follow up ticket